match/lib/match/runner.rb

```diff
@@ -166,7 +166,7 @@ def fetch_provisioning_profile(params: nil, certificate_id: nil, app_identifier:
 
       if params[:force_for_new_devices] && !params[:readonly]
         if prov_type != :appstore
-          params[:force] = device_count_different?(profile: profile, keychain_path: keychain_path) unless params[:force]
+          params[:force] = device_count_different?(profile: profile, keychain_path: keychain_path, platform: params[:platform].to_s) unless params[:force]
```

Comment:
Can we remove this `.to_s` here? And all the other `.to_s`s inside of the case below? 🤔 Should be able to case on just the symbols 👌

Reply:
@joshdholtz: I'm mostly writing Swift these days—so please keep any and all feedback coming!

Regarding your comment, not sure we can drop the string conversion. I tried the following and all matching squashed down to the `else` clause of the switch. Any suggestions?

```patch
diff --git a/match/lib/match/runner.rb b/match/lib/match/runner.rb
index 5f7c7eacd..e3fdeba19 100644
--- a/match/lib/match/runner.rb
+++ b/match/lib/match/runner.rb
@@ -166,7 +166,7 @@ module Match

       if params[:force_for_new_devices] && !params[:readonly]
         if prov_type != :appstore
-          params[:force] = device_count_different?(profile: profile, keychain_path: keychain_path, platform: params[:platform].to_s) unless params[:force]
+          params[:force] = device_count_different?(profile: profile, keychain_path: keychain_path, platform: params[:platform]) unless params[:force]
         else
           # App Store provisioning profiles don't contain device identifiers and
           # thus shouldn't be renewed if the device count has changed.
@@ -241,11 +241,11 @@ module Match

         portal_device_count =
           case platform
-          when :ios.to_s
+          when :ios
             Spaceship.device.all_ios_profile_devices.count
-          when :tvos.to_s
+          when :tvos
             Spaceship.device.all_apple_tvs.count
-          when :mac.to_s
+          when :mac
             Spaceship.device.all_macs.count
           else
             Spaceship.device.all.count
```